src/HttpRequest.cc

```diff
@@ -738,3 +738,31 @@ HttpRequest::manager(const CbcPointer<ConnStateData> &aMgr, const AccessLogEntry
     }
 }
 
+const Ip::Address *
+FindListeningPortAddress(const HttpRequest *request, const bool derive_ips, const Ip::Address *cacheip, const Ip::Address *localip)
+{
+    if (request) {
+        if (request->flags.interceptTproxy || request->flags.intercepted) {
+            if (!cacheip && derive_ips && request->masterXaction->squidPort)
+                cacheip = &(request->masterXaction->squidPort->s);
+            if (cacheip)
+                return (cacheip->isAnyAddr() ? nullptr : cacheip);
+        }
+        if (!localip && derive_ips && request->masterXaction->tcpClient)
+            return &(request->masterXaction->tcpClient->local);
+    }
+    return localip;
+}
+
+const Ip::Address *FindListeningPortAddress(const HttpRequestPointer &request)
```

Comment:
Modern Squid code puts return type in not-inlined function definitions on a separate line.